### DISCUSSION TITLE
Artifact bindings/java/dist/sleuthkit-4.x.y.jar not created on Linux

### DISCUSSION THREAD

**Discussion by afonsohca** [sepinf-inc/IPED#1240] [Q&A]
Este é meu sistema operacional: https://i.imgur.com/OsY8R8v.png
Ubuntu 20.04.4 LTS 64bits.

Para instalar e usar a ferramenta no meu sistema operacional, estou seguindo a documentação deste link: https://github.com/sepinf-inc/IPED/wiki/Linux

Consegui instalar todas as dependências necessárias, como se vê:
Java: https://i.imgur.com/OWMSiR9.png

Quanto ao sleuthkit, acredito que o erro resida justamente ai. Ao final do download, após dar os comandos $./bootstrap e $./configure, verifiquei o seguinte resultado (no final, como se vê no print):

> Building:
>    afflib support:                        yes
>    libewf support:                        yes
>    zlib support:                          yes
> 
>    libvhdi support:                       no
>    libvmdk support:                       yes
> Features:
>    Java/JNI support:                      no
>    Multithreading:                        yes

https://i.imgur.com/AkIxTRN.png

Acredito que o erro resida justamente nesse não-suporte ao java e ao libvhdi...não estou conseguindo resolver sozinho isso.

Quando em seguida vou para os comandos $make e $make install, concluo que o sleuthkit foi enfim compilado, como se vê: https://i.imgur.com/9ifg1TS.png

Versão 4.6.0

Na documentação, fala que após estes comandos, deveria surgir o elemento **bindings/java/dist/sleuthkit-4.xyjar**, porém, ele não foi criado na instalação!!! E este arquivo é necessário pois seu caminho precisa ser adicionado no LocalConfig do IPED...

Estou travado nisso pessoal!!! Sei que pode parecer simples, mas, não estou conseguindo avançar na instalação do software no ubuntu...se alguém puder me ajudar, serei muito grato!!! Desde já, obrigado.

**Comment by paulobreim** [comment]
Já tentou usar o docker do iped?

**Comment by lfcnassif** [comment]
Have you installed ant and configured JAVA_HOME env var?

**Reply by lfcnassif** [reply]
I just looked at your screens, you don't have `ant`

**Reply by afonsohca** [reply]
> Você já instalou formigas e configurou JAVA_HOME env var?

Não...confesso que não sabia!! 
https://receitasdecodigo.com.br/ubuntu/como-configurar-java_home-para-java-no-ubuntu
Achei este tutorial, vou tentar seguir com a instalação...

**Reply by afonsohca** [reply]
Nassif, veja:
https://i.imgur.com/32FyT2D.png
Porém, o suporte ao Java continua marcado como "no" depois do ./configure

**Reply by lfcnassif** [reply]
You should install `ant`

**Reply by afonsohca** [reply]
Deu certo!! Consegui compilar perfeitamente o sleuthkit e segui o passo a passo...alterei o LocalConfig do IPED com o endereço do artefato que surgiu...
Mas, não estou conseguindo executar, meu SO não está localizando o iped.jar, veja:
https://i.imgur.com/U7Kuab4.png
Desde já muito obrigado por sua ajuda, amigo. Desculpa se as perguntas são leigas demais...

**Reply by afonsohca** [reply]
Baixei a versão 3.18 e dessa vez, consegui...porém, surgiu este erro:
https://i.imgur.com/8VGypfa.png

**Reply by paulobreim** [reply]
@afonsohca você pode colocar a imagem no próprio texto, não precisa usar outro site. veja:

![image](https://user-images.githubusercontent.com/381594/181823510-c0d30ec5-4653-4a84-ae2b-143f932a1ce7.png)

**Reply by afonsohca** [reply]
> @afonsohca você pode colocar a imagem no próprio texto, não precisa usar outro site. veja:
> 
> ![image](https://user-images.githubusercontent.com/381594/181823510-c0d30ec5-4653-4a84-ae2b-143f932a1ce7.png)

Boa!!! Não sabia...

**Comment by afonsohca** [comment]
![image](https://user-images.githubusercontent.com/110132252/181836546-8e735758-3600-4d93-b898-1db8f4115cdd.png)

Consegui resolver essa lib que estava faltando...mas, estranhamente o iped não roda por falta de memória...minha máquina tem 16GB de RAM.

**Reply by lfcnassif** [reply]
Have you tried all 3 suggested options in dialog above?

**Reply by lfcnassif** [reply]
By the way, you are using a very outdated version not being maintained anymore.

**Reply by afonsohca** [reply]
> Você já experimentou todas as 3 opções sugeridas em diálogo acima?

sim, todas...

![image](https://user-images.githubusercontent.com/110132252/182161849-fe048764-3edf-4a23-812e-cd47c0a80575.png)

**Reply by afonsohca** [reply]
> > Você já experimentou todas as 3 opções sugeridas em diálogo acima?
> 
> sim, todas...
> 
> ![image](https://user-images.githubusercontent.com/110132252/182161849-fe048764-3edf-4a23-812e-cd47c0a80575.png)

Depois disso, desinstalei o java e reinstalei pra uma versão 64bits.

**Reply by lfcnassif** [reply]
2GB is too little. Since you have 16GB of RAM, I recommend using half as a rule of thumb.

**Reply by afonsohca** [reply]
> que você tem 16GB de RAM, recomendo usar m

Eu tentei, mas só consegui fazer rodar com 2GB..
![image](https://user-images.githubusercontent.com/110132252/182178712-a1bdf1ec-bed8-4a9a-a581-0b9b557b258f.png)

**Reply by afonsohca** [reply]
> > que você tem 16GB de RAM, recomendo usar m
> 
> Eu tentei, mas só consegui fazer rodar com 2GB.. ![image](https://user-images.githubusercontent.com/110132252/182178712-a1bdf1ec-bed8-4a9a-a581-0b9b557b258f.png)

Veja:

![image](https://user-images.githubusercontent.com/110132252/182179510-6ad2da1c-febf-470e-ac9b-8ef1d9b4f9da.png)

**Reply by lfcnassif** [reply]
You are still using java 32bits, not 64bits...

**Comment by patrickdalla** [comment]
emperrei na compilação do java binding do sleuth na confoguração do proxy.
o Ant não reconhece o proxy das variáveis de ambiente. tem que confifueá-lo
alterando o arquivo build.xml

Em qua., 27 de jul. de 2022 17:12, afonsohca ***@***.***>
escreveu:
